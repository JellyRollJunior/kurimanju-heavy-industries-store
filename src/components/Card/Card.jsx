import { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.css';

const Card = ({ id, name, price, img, onSubmit }) => {
  const [quantity, setQuantity] = useState(0);
  
  const onSubmitCart = (event) => {
    event.preventDefault();
    onSubmit(id, quantity);
  }

  return (
    <div className={styles.card}>
      <img src={img} alt="" />
      <h3>{name}</h3>
      <h4>{price}</h4>
      <form onSubmit={(event) => onSubmitCart(event)}>
        <input type="number" value={quantity} onChange={(event) => setQuantity(event.target.value)} required />
        <button>Add to Cart</button>
      </form>
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  price: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  img: PropTypes.string,
}

export { Card };
