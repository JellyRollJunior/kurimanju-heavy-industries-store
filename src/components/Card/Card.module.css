.card {
  composes: secondaryColor from '../../styles/colors.module.css';
  composes: primaryBackground from '../../styles/colors.module.css';
  display: flex;
  flex-direction: column;
  padding: 24px 16px 14px 16px;

  text-align: right;
  border: 2px solid;
  border-radius: 4px;
}

.form {
  composes: primaryColor from '../../styles/colors.module.css';
  display: flex;
  flex-direction: column;
}

.form > * {
  display: block;
  margin-left: auto;
}

.quantity {
  max-width: 80px;
  margin-top: 6px;
  padding: 3px 3px 1px 3px;

  text-align: right;
  color: inherit;
  background-color: inherit;
  border: 2px dashed;
  border-radius: 6px;
}

.submitBtn {
  composes: quaternaryColor from '../../styles/colors.module.css';
  margin-top: 10px;
  padding: 3px 18px 1px 18px;

  background-color: inherit;
  border: 2px solid;
  border-radius: 4px;
}

.itemName {
  composes: tertiaryColor from '../../styles/colors.module.css';
}