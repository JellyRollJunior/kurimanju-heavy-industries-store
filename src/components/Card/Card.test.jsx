import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';
import { Card } from './Card';
import { render, screen } from '@testing-library/react';

describe('Shop product card', () => {
  it('Renders produce image, title, price, quantity controls, and an add to cart button', () => {
    const { container } = render(<Card />);

    expect(container).toMatchSnapshot();
  })

  it('Calls onSubmit on clicking add to cart button', async () => {
    const onSubmit = vi.fn();
    const user = userEvent.setup();
    render(<Card onSubmit={onSubmit} />);
    const button = screen.getByRole('button', { name: 'Add to Cart' });

    await user.click(button);

    expect(onSubmit).toHaveBeenCalled();
  });
});
