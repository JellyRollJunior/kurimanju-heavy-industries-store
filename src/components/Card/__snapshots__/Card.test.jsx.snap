// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop product card > Renders produce image, title, price, quantity controls, and an add to cart button 1`] = `
<div>
  <div
    class="_card_1f2933"
  >
    <img
      alt="Unable to retrieve product info"
      src="/src/assets/hero.png"
    />
    <form
      class="_form_1f2933"
    >
      <h3
        class="_itemName_1f2933"
      >
        Unable to retrieve product info
      </h3>
      <h4 />
      <div
        class="_quantityHolder_1f2933"
      >
        <button
          class="_quantityButton_1f2933 _decrementButton_1f2933"
        >
          -
        </button>
        <input
          class="_quantity_1f2933"
          disabled=""
          required=""
          type="number"
          value="0"
        />
        <button
          class="_quantityButton_1f2933 _incrementButton_1f2933"
        >
          +
        </button>
      </div>
      <button
        class="_submitBtn_1f2933"
      >
        Add to Cart
      </button>
    </form>
  </div>
</div>
`;
